### ISSUE TITLE
contract void reinstatment not debiting finance and insurance charges

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1367]
_Imported from trac ticket 1367.
Created by **rneville**
Opened in trac: Jan 16, 2012, 10:50:49 AM
Last modified in trac: Jan 17, 2012, 10:48:05 AM_
***

**Comment by kbower** [comment]
_**rneville** trac comment on Jan 16, 2012, 10:51:21 AM:_

Yes, basically every transaction you do to payoff the old contract needs to be reinstated back to the old contract when the new contract is voided.

Sal

Sent from my iPhone

On Jan 13, 2012, at 1:11 PM, Rob Neville <rn@retailarchitects.com> wrote:

OK, I'm confused....(not hard to do ;) )

When we do an addon, we calculate the payoff amount which is Credited to the old account via a DTC and we then credit the earned portions of finance charges and insurance charges through FIC and IxC transactions bringing the old contract's balance to zero.

So I think I understand what you're saying but I want to verify...I'm debiting the old account properly with a DTD to wash out the original DTC, but what I'm not doing is balancing out those "unearned" credits that we did to pay it off. That is to say, I should be issuing FID and IxD transactions to offset the credits I did when it was added on.

Let me put that into an illustration and tell me if I've got it right....

old contract
SAL 1000
FID 100
IAD 20

added on to new contract:
old contract
SAL 1000
FID 100
IAD 20
FIC 10
IAC 5
DTC 1115

new contract
SAL 500
DTD 1115
FID 200
IAD 20

Now voiding the new contract should do
new contract
SAL 500
DTD 1115
FID 200
IAD 20
DTC 1115
FIC 200
IAD 20

old contract
SAL 1000
FID 100
IAD 20
FIC 10
IAC 5
DTC 1115
DTD 1115
FID 10
IAD 5

Bolded ones are what you're saying I'm missing??

On Jan 13, 2012, at 1:11 PM, Sal Genovese wrote:

The original contract did have FIC, etc but once the new contract was voided, these transaction did not get reinstated. Only the dollar transfer got reinstated 

Sent from my iPhone

On Jan 13, 2012, at 12:05 PM, Rob Neville <rn@retailarchitects.com> wrote:

So you voided a contract that had an original contract and the original contract did not have FIC, IxC etc?

On Jan 13, 2012, at 1:00 PM, Sal Genovese wrote:

Rob-

We getting closer, but it's still not reinstating the FIC, IAC, ICC transactions. It is doing the DTC reinstate.

Sent from my iPhone
